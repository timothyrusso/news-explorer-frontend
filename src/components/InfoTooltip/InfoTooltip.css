.info-tooltip__text {
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 900;
  font-size: 24px;
  line-height: 28px;
  color: #000;
  margin: 43px 79px 14px 36px;
}

.info-tooltip__button {
  font-family: 'Inter', Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 22px;
  border: none;
  background-color: #fff;
  color: #2F71E5;
  margin: 0 166px 51px 36px;
}
