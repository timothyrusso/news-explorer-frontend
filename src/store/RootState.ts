import { UserState } from './user/user.reducer';
import { ArticleState } from './article/article.reducer';
import { TogglesState } from './toggles/toggles.reducer';
import { ErrorsState } from './errors/errors.reducer';

export type RootState = {
  user: UserState;
  article: ArticleState;
  toggles: TogglesState;
  errors: ErrorsState;
};
