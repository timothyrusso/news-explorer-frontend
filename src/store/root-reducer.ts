import { combineReducers } from 'redux';
import { userReducer } from './user/user.reducer';
import { articleReducer } from './article/article.reducer';

export const rootReducer = combineReducers({
  user: userReducer,
  article: articleReducer,
});
