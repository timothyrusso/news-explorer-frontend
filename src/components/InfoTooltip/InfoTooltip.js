import "./InfoTooltip.css";
import Popup from "../Popup/Popup";

const InfoTooltip = ({ isOpen, onClose, openSignin }) => {
  return (
    <Popup isOpen={isOpen} onClose={onClose} infoTooltip={true}>
      <div className="info-tooltip">
        <p className="info-tooltip__text">
          Registration successfully completed!
        </p>
        <button
          aria-label="signin"
          type="button"
          className="info-tooltip__button"
          onClick={openSignin}
        >
          Sign in
        </button>
      </div>
    </Popup>
  );
};

export default InfoTooltip;
