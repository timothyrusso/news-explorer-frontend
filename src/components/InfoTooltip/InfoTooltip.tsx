import React from 'react';
import Popup from '../Popup/Popup';
import { usePopup } from '../../hooks/usePopup';
import { useSelector } from 'react-redux';
import { RootState } from '../../store/RootState';
import './InfoTooltip.css';

const InfoTooltip = () => {
  const { handleSigninPopupClick } = usePopup();
  const isOpen = useSelector(
    (state: RootState) => state.toggles.isInfoTooltipOpen
  );

  return (
    <Popup isOpen={isOpen} infoTooltip={true}>
      <div className="info-tooltip">
        <p className="info-tooltip__text">
          Registration successfully completed!
        </p>
        <button
          aria-label="signin"
          type="button"
          className="info-tooltip__button"
          onClick={handleSigninPopupClick}
        >
          Sign in
        </button>
      </div>
    </Popup>
  );
};

export default InfoTooltip;
