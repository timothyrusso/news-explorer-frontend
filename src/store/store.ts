import { legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { rootReducer } from './root-reducer';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { RootState } from './RootState';

const store = createStore(rootReducer, applyMiddleware(thunk));

export type AppDispatch = ThunkDispatch<RootState, unknown, AnyAction>;

export default store;
