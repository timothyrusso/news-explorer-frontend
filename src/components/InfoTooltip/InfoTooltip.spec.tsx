import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store/store';
import InfoTooltip from './InfoTooltip';

describe('InfoTooltip component', () => {
  it('Should render the InfoTooltip component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <InfoTooltip />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
