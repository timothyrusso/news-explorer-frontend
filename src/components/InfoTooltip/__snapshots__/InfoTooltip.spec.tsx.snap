// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTooltip component Should render the InfoTooltip component 1`] = `
<DocumentFragment>
  <div
    class="popup  popup_type_undefined"
  >
    <div
      class="popup__container"
    >
      <div
        class="info-tooltip"
      >
        <p
          class="info-tooltip__text"
        >
          Registration successfully completed!
        </p>
        <button
          aria-label="signin"
          class="info-tooltip__button"
          type="button"
        >
          Sign in
        </button>
      </div>
      <button
        aria-label="Close undefined popup"
        class="close-button close-button_type_info"
        type="button"
      />
    </div>
  </div>
</DocumentFragment>
`;
